thread.c

```diff
@@ -689,29 +711,36 @@ void dispatch_conn_new(int sfd, enum conn_states init_state, int event_flags,
     item->mode = queue_new_conn;
     item->ssl = ssl;
 
-    cq_push(thread->new_conn_queue, item);
-
     MEMCACHED_CONN_DISPATCH(sfd, (int64_t)thread->thread_id);
-    buf[0] = 'c';
-    if (write(thread->notify_send_fd, buf, 1) != 1) {
-        perror("Writing to thread notify pipe");
-    }
+    notify_worker(thread, item);
 }
 
 /*
  * Re-dispatches a connection back to the original thread. Can be called from
  * any side thread borrowing a connection.
  */
-#define REDISPATCH_MSG_SIZE (1 + sizeof(int))
 void redispatch_conn(conn *c) {
-    char buf[REDISPATCH_MSG_SIZE];
-    LIBEVENT_THREAD *thread = c->thread;
+    notify_worker_fd(c->thread, c->sfd, queue_redispatch);
+}
 
-    buf[0] = 'r';
-    memcpy(&buf[1], &c->sfd, sizeof(int));
-    if (write(thread->notify_send_fd, buf, REDISPATCH_MSG_SIZE) != REDISPATCH_MSG_SIZE) {
-        perror("Writing redispatch to thread notify pipe");
+void timeout_conn(conn *c) {
```

Comment:
Note: would prefer exposing the enum and having 1-2 public functions for notifying a worker of something but don't feel like I need to do this now.